pyvcloud/vcd/vdc.py

```diff
@@ -484,18 +484,29 @@ def get_disk(self, name, disk_id=None):
 
         disks = self.get_disks()
 
+        result = None
         if disk_id is not None:
             if not disk_id.startswith('urn:vcloud:disk:'):
                 disk_id = 'urn:vcloud:disk:' + disk_id
             for disk in disks:
                 if disk.get('id') == disk_id:
-                    return disk
+                    result = disk
+                    # disk-id's are unique so it's ok to break the loop
+                    # and stop looking further.
+                    break
+        elif name is not None:
+            for disk in disks:
+                if disk.get('name') == name:
+                    if result is None:
+                        result = disk
+                    else:
+                        raise Exception('Found multiple disks with name %s'
```

Comment:
Did you make sure all the existing commands (vcd-cli) invoking get_disk(..) have an optional parameter to pass the disk_id?
ex: attachdisk, detach disk, update disk etc.

If not, please fix them as part of this change.

Reply:
Yes, all of them has an optional param 'id'

Reply:
Wait, update, change owner and delete disk has the optional field id. Attach and detach don't have it.  I will add them. Btw the changes will be in vcd-cli and there won't be any further changes in this pull request.

Reply:
okay, approving this change, will expect review on vcd-cli